app/controllers/admin/territories/organisations_controller.rb

```diff
@@ -0,0 +1,82 @@
+class Admin::Territories::OrganisationsController < Admin::Territories::BaseController
+  def new
+    @organisation = Organisation.new(territory: current_territory)
+    authorize(@organisation, policy_class: Agent::OrganisationPolicy)
+  end
+
+  def create
+    @organisation = Organisation.new(
+      agent_roles_attributes: [{ agent: current_agent, access_level: AgentRole::ACCESS_LEVEL_ADMIN }],
+      verticale: current_domain.verticale,
+      territory: current_territory,
+      **params.require(:organisation).permit(:name)
+    )
+    authorize(@organisation, policy_class: Agent::OrganisationPolicy)
+
+    if @organisation.save
+      redirect_to admin_organisation_configuration_path(@organisation),
+                  flash: { success: "Organisation enregistrée ! Vous pouvez maintenant lui ajouter des motifs et des lieux de rendez-vous, puis inviter des agents à la rejoindre" }
+    else
+      render :new
+    end
+  end
+
+  def index
+    @organisations = organisations.where(closed_at: nil)
+    @closed_organisations = current_territory.organisations.where.not(closed_at: nil)
+  end
+
+  def close
+    organisation = Organisation.find(params[:id])
+    authorize(organisation, :close?, policy_class: Agent::OrganisationPolicy)
+
+    organisation.agents.where.not(id: current_agent.id).each do |agent|
+      agent_removal = AgentRemoval.new(agent, organisation)
+      if agent_removal.valid?
+        agent_removal.remove!
+      end
+    end
+
+    if organisation.agents == [current_agent]
+      agent_removal = AgentRemoval.new(current_agent, organisation)
+      agent_removal.remove! if agent_removal.valid?
+    end
+
+    if organisation.reload.agents.empty?
+      organisation.update!(closed_at: Time.zone.now)
+      flash[:success] = "L'organisation a été fermée."
+    else
+      flash[:error] = "L'organisation n'a pas pu être fermée parce que des agents on encore des rendez-vous à venir dans cette organisation."
```

Comment:
je mettrais bien un peu d’hypothétique dans cette phrase car potentiellement c’est une autre raison qui a empêché le retrait de l’agent de l’orga. 

```suggestion
      flash[:error] = "L'organisation n'a pas pu être fermée. Une raison qui peut bloquer est qu’il y a des RDV à venir dans cette organisation."
```

Reply:
D'après ce que je comprends du code de AgentRemoval, aujourd'hui la seule raison possible est qu'un agent ai des rendez-vous à venir.
C'est possible que le code change, et que ça devienne faux, mais d'ici là, je préfère éviter de formuler des manière hypothétique des informations.